### PR TITLE
Fixes #18106: Context should contain only variable->type mapping

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/18106

### CODE REVIEW

File: rudder-lang/src/ast.rs [hunk 1]
```diff
@@ -171,16 +172,16 @@ impl<'src> AST<'src> {
...
                 variable,
-                Value::from_static_pvalue(value.clone()).unwrap(),
+                &Value::from_static_pvalue(value.clone()).unwrap(),
             ) {
                 self.errors.push(e);
             }
             let getter = |k| self.context.get(&k);
             match Value::from_pvalue(&self.enum_list, &getter, value) {
-                Err(e) => self.errors.push(e),
+                Err(e) => { self.errors.push(e); },
```
Comment (gpoblon) [inline rudder-lang/src/ast.rs:184]: can't it be inline?

File: rudder-lang/src/ast.rs [hunk 2]
```diff
@@ -171,16 +172,16 @@ impl<'src> AST<'src> {
...
                 Ok(val) => {
                     self.variable_declarations.insert(variable, val);
```
Comment (gpoblon) [inline rudder-lang/src/ast.rs:186]: here too

**Comment by peckpeck** [comment]
PR updated with a new commit

File: rudder-lang/src/ast.rs (hunk 2 above)
Comment (peckpeck) [reply rudder-lang/src/ast.rs:186]: not this one, it needs a ';' to ignore return value

**Comment by peckpeck** [comment]
OK, squash merging this PR

**Pull request merged**